### ISSUE TITLE
localStorage is cleared when popup window for same domain is open

### ISSUE THREAD

**Issue by dennmart** [DevExpress/testcafe#5474]
### What is your Test Scenario?
I built a small application that uses localStorage for some data. The app has one main window and a popup window in the same domain, so they share the same localStorage data.

In my test scenario, I want to load the main window first, click on a button that opens the popup window, and run some scenarios from within the popup window, which uses the localStorage data.
 
### What is the Current behavior?
In TestCafe 1.9.1, when I click on a link or button that opens a separate window, it's clearing the localStorage data even though the data was already set in the main screen and the popup window shares the same domain.
 
### What is the Expected behavior?
It should behave the same as TestCafe 1.8.7, where opening a separate window based on an action (like clicking a link or button) for the same domain / website doesn't clear localStorage.
 
### What is your web application and your TestCafe test code?

Test example:

 
```js
import { ClientFunction } from "testcafe";

fixture("Simple Shopping List").page("https://shoppinglist.dev-tester.com");

test("Add and remove items from empty shopping list", async (t) => {
  // When the home page loads, it will check for the existence of the 
  // `shoppingListId` key in localStorage, and automatically generate
  // set a UUID if it's null.
  const id = ClientFunction(() => localStorage.getItem("shoppingListId"));
  // Prints out the localStorage data, automatically generated by the app when loading the page.
  console.log(await id());

  // Opens the popup window.
  await t.click("#add_new_items_btn");

  // In TestCafe 1.8.7, this prints out the same ID as the previous console.log.
  // In TestCafe 1.9.1, this prints a different ID since the app's JavaScript automatically
  // generates an ID when the localStorage data doesn't exist.
  console.log(await id());
});

```

<details>
<summary>Your complete test report:</summary>

```
# TestCafe 1.8.7:

mac@Dennis-Mac simple_shopping_list_tests % testcafe chrome list_test.js
Using locally installed version of TestCafe.
 Running tests in:
 - Chrome 85.0.4183.83 / macOS 10.15.6

 Simple Shopping List
b534cc71-1ed1-4a65-97e4-e32934208ad4
b534cc71-1ed1-4a65-97e4-e32934208ad4
 ✓ Add and remove items from empty shopping list


 1 passed (1s)
```

```
# TestCafe 1.9.1:

mac@Dennis-Mac simple_shopping_list_tests % testcafe chrome list_test.js
Using locally installed version of TestCafe.
 Running tests in:
 - Chrome 85.0.4183.83 / macOS 10.15.6

 Simple Shopping List
40ba38d9-0a23-4871-9af3-b592dca9eaec
4562e3ec-08e2-4e98-9b3f-b03ca292c581
 ✓ Add and remove items from empty shopping list


 1 passed (1s)
```
</details>
 
### Your Environment details:
 
* testcafe version: 1.9.1                   
* node.js version: v12.16.1                   
* command-line arguments: `testcafe chrome list_test.js`  
* browser name and version: Google Chrome 85.0.4183.83 
* platform and version: macOS 10.15.6

**Comment by Ogurecher** [comment]
@dennmart, Thank you for your report. I've reproduced this issue. We'll update this thread once it's fixed.
In the meantime, you can run TestCafe with the `--disable-multiple-windows` flag as a workaround.

**Comment by Farfurix** [comment]
Hello,

The page is not available and I see no examples here. Could you please share [some example](https://testcafe.io/402636/faq#how-to-create-a-minimal-working-example-when-you-submit-an-issue) if this issue is still reproducible?

**Comment by dennmart** [comment]
Hi @Farfurix,

Sorry, I had taken the app down since the original issue had been confirmed over a year ago. I verified locally with the latest releases of TestCafe and the originally reported issue seems to be fixed now.

Out of curiosity I checked older versions and the problem had been present from v1.9.0 all the way up to v1.16.1. The problem was corrected with TestCafe 1.17.0. Not sure what the exact fix was since there's nothing in the changelogs that addressed this issue directly, but at least it seems to be fine now.

Thanks to you and @Ogurecher for initially looking into this!
